app/celery/tasks.py

```diff
@@ -110,36 +113,53 @@ def copy_s3_object(source_bucket, source_filename, target_bucket, target_filenam
     )
 
 
-@notify_celery.task(
-    bind=True,
-    name="create-pdf-for-templated-letter",
-    max_retries=3,
-    default_retry_delay=180,
-)
-def create_pdf_for_templated_letter(self, encrypted_letter_data):
-    letter_details = current_app.encryption_client.decrypt(encrypted_letter_data)
-    current_app.logger.info("Creating a pdf for notification with id %s", letter_details["notification_id"])
+def _create_pdf_for_letter(task: Task, letter_details, language: Literal["english", "welsh"]):
     logo_filename = f'{letter_details["logo_filename"]}.svg' if letter_details["logo_filename"] else None
-
     template = LetterPrintTemplate(
         letter_details["template"],
         values=letter_details["values"] or None,
         contact_block=letter_details["letter_contact_block"],
         # letter assets are hosted on s3
         admin_base_url=current_app.config["LETTER_LOGO_URL"],
         logo_file_name=logo_filename,
+        language=language,
     )
     with current_app.test_request_context(""):
         html = HTML(string=str(template))
 
     try:
-        with sentry_sdk.start_span(op="function", description="weasyprint.HTML.write_pdf"):
+        with sentry_sdk.start_span(op="function", description="weasyprint.HTML.write_pdf[english]"):
```

Comment:
should the language here be passed in, instead of hard-coded to [english]?

Reply:
oops yes 🙌